lib/nerves_hub/scripts/script.ex

```diff
@@ -3,23 +3,36 @@ defmodule NervesHub.Scripts.Script do
 
   import Ecto.Changeset
 
+  alias NervesHub.Accounts.User
   alias NervesHub.Products.Product
 
   @type t :: %__MODULE__{}
+  @required [:name, :text]
 
   schema "scripts" do
     belongs_to(:product, Product)
+    belongs_to(:created_by, User, where: [deleted_at: nil])
+    belongs_to(:last_updated_by, User, where: [deleted_at: nil])
 
     field(:name, :string)
     field(:text, :string)
 
     timestamps()
   end
 
-  def changeset(struct, params) do
+  def create_changeset(struct, params) do
     struct
-    |> cast(params, [:name, :text])
-    |> validate_required([:name, :text])
+    |> cast(params, @required ++ [:created_by_id])
+    |> validate_required(@required ++ [:created_by_id])
     |> validate_length(:name, lte: 255)
+    |> foreign_key_constraint(:created_by_id)
+  end
+
+  def update_changeset(struct, params) do
```

Comment:
```suggestion
  def update_changeset(struct, params \\ %{}) do
```
If we add a sensible default here we don't have to remember to pass in an empty map when calling. Small nit for me, feel free to disregard.